### 1. Conditional Basics

a. prompt the user for a day of the week, print out whether the day is Monday or not

In [1]:
day_of_the_week = input('Day of the Week: ')
if day_of_the_week == 'Monday' or day_of_the_week == 'monday':
    print("It's another wonderful Monday!")
else:
    print(f'{day_of_the_week} is just another day of the week.')

Day of the Week: Doomsday
Doomsday is just another day of the week.


b. prompt the user for a day of the week, print out whether the day is a weekday or a weekend

In [2]:
weekday = ['Monday', 'Mon', 'Tuesday', 'Tues', 'Wednesday', 'Wed', 'Thursday', 'Thur', 'Friday', 'Fri']
weekend = ['Saterday', 'Sat', 'Sunday', 'Sun']
day_of_the_week = input('Day of the Week: ')
if day_of_the_week in weekday:
    print(f'{day_of_the_week} is a weekday.')
elif day_of_the_week in weekend:
    print(f'{day_of_the_week} is on the weekend.')
else:
    print(f'{day_of_the_week} is not a valid day of the week!')

Day of the Week: Doomsday
Doomsday is not a valid day of the week!


c. create variables and make up values for

* the number of hours worked in one week
* the hourly rate
* how much the week's paycheck will be

write the python code that calculates the weekly paycheck. 
You get paid time and a half if you work more than 40 hours.

In [3]:
weekly_hours_worked = 60
hourly_wage = 10
weekly_paycheck = weekly_hours_worked * hourly_wage
overtime = hourly_wage * 1.5

if weekly_hours_worked <= 40: 
    print(weekly_paycheck)

elif weekly_hours_worked > 40: 
    print(weekly_hours_worked * overtime)

900.0


### 2. Loop Basics

a. While

* Create an integer variable i with a value of 5.
* Create a while loop that runs so long as i is less than or equal to 15
* Each loop iteration, output the current value of i, then increment i by one.

Your output should look like this:

5
6
7
8
9
10
11
12
13
14
15

In [4]:
i = int(5)
while i <= 15:
    print(i) # order of print matters
    i += 1

5
6
7
8
9
10
11
12
13
14
15


* Create a while loop that will count by 2's starting with 0 and ending at 100. Follow each number with a new line.

In [7]:
i = 0

while i <= 100:
    print(i)
    i += 2

0
2
4
6
8
10
12
14
16
18
20
22
24
26
28
30
32
34
36
38
40
42
44
46
48
50
52
54
56
58
60
62
64
66
68
70
72
74
76
78
80
82
84
86
88
90
92
94
96
98
100


* Alter your loop to count backwards by 5's from 100 to -10.

In [1]:
i = 100

while i >= -10 and i <= 100: # order is important!
    # less than 100 listed first will NOT stop at -10
    print(i)
    i -= 5

100
95
90
85
80
75
70
65
60
55
50
45
40
35
30
25
20
15
10
5
0
-5
-10


* Create a while loop that starts at 2, and displays the number squared on each line while the number is less than 1,000,000. 

Output should equal:

 2
 4
 16
 256
 65536

In [3]:
number = 2
while number < 1000000:
    print(number)
    number **= 2

2
4
16
256
65536


Write a loop that uses print to create the output shown below.

100
95
90
85
80
75
70
65
60
55
50
45
40
35
30
25
20
15
10
5

In [4]:
i = 100

while i >= 5 and i <= 100:
    print(i)
    i -= 5

100
95
90
85
80
75
70
65
60
55
50
45
40
35
30
25
20
15
10
5


b. For Loops

i. Write some code that prompts the user for a number, then shows a multiplication table up through 10 for that number.

For example, if the user enters 7, your program should output:

7 x 1 = 7
7 x 2 = 14
7 x 3 = 21
7 x 4 = 28
7 x 5 = 35
7 x 6 = 42
7 x 7 = 49
7 x 8 = 56
7 x 9 = 63
7 x 10 = 70

In [25]:
multipliers = map(int, list(range(1,11)))
number = int(input('Choose a whole number: '))
for multiplier in multipliers:
    result = number * multiplier
    print(f'{number} x {multiplier} = {result}')

Choose a whole number: 7
7 x 1 = 7
7 x 2 = 14
7 x 3 = 21
7 x 4 = 28
7 x 5 = 35
7 x 6 = 42
7 x 7 = 49
7 x 8 = 56
7 x 9 = 63
7 x 10 = 70


NOTE: 1st attempt grabbed input as str

In [31]:
a = input('hg')
for i in range(1,11):
    print(f'{a} * {i} = {a*i}')

hg7
7 * 1 = 7
7 * 2 = 77
7 * 3 = 777
7 * 4 = 7777
7 * 5 = 77777
7 * 6 = 777777
7 * 7 = 7777777
7 * 8 = 77777777
7 * 9 = 777777777
7 * 10 = 7777777777


ii. Create a for loop that uses print to create the output shown below.


1
22
333
4444
55555
666666
7777777
88888888
999999999

In [42]:
a = 1 # a not needed i * i would work
for i in range(1,10):
    print(str(a)*i)
    a += 1

1
22
333
4444
55555
666666
7777777
88888888
999999999
